sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2491,7 +2499,7 @@ def is_provided_ips_valid(ips: Optional[List[Optional[str]]]) -> bool:
                 handle=self,
                 head_ip_max_attempts=max_attempts,
                 worker_ip_max_attempts=max_attempts,
-                get_internal_ips=False)
+                get_internal_ips=use_internal_ips)
```

Comment:
Q: is this a bug? If so, this should've affected AWS internal IP-only users. Why was it not triggered?

Naming: can be confusing to call the result cluster_external_ips if it can be internal.